### DISCUSSION TITLE
I'd like "Ctrl+A" to slowly expand the code selection, rather than highlight everything

### DISCUSSION THREAD

**Discussion by StachuDotNet** [darklang/dark#4597] [Suggestions]
In [Dynalist](https://dynalist.io/), Ctrl-A slowly expands the scope of what I'm currently selecting rather than selecting _all_ text available.

![Peek 2022-11-10 19-11](https://user-images.githubusercontent.com/906686/201232361-f618d9ae-3f80-45e9-9fb4-51f40e8a6e85.gif)

I _love_ this feature. It makes it super ergonomic to copy and manipulate the structured text.
I think this would be appropriate for Dark as well - each Ctrl+A would expand just by one layer (pattern/expr/etc).
Even better, this could work on a `AstRef` level (e.g. so a ctrl+a on `1|2.34` would first just highlight `12`)

**Comment by pbiggar** [comment]
I've wanted something like this for a while - paredit, on which our concept of the structured editor is based, has similar capabilities.

I don't want to use ctrl-a for it though - that already has a standard (select-all) meaning.
